### DISCUSSION TITLE
Creating a SELECT query that consumes a tuple where casting is required. Is it possible to do this without using rawQuery?

### DISCUSSION THREAD

**Discussion by MeoMix** [prisma/prisma#15071] [Q&A] [answered]
I have the following code which works well:

```
      const idLastUpdatedPairs = orderItemIdLastUpdatedTuples
        .map(([id, lastUpdated]) => `('${id}'::uuid, '${lastUpdated.toISOString()}')`)
        .join(', ');
      const query = `SELECT * FROM order_item WHERE (id, last_updated) IN (${idLastUpdatedPairs})`;
      const items = await prisma.$queryRaw<OrderItemDataAccessModel[]>(Prisma.raw(query));
```

I am interested in more succinct representations of it. Ideally not using rawQuery and leveraging Prisma directly.

I am mapping and joining manually because Prisma.join() won't consume an array of UUIDs in such a way that it casts them. If that issue were overcome then I believe it might also struggle with the tuple syntax, but am uncertain.

I wasn't able to surface a way to convey the uuid type to Prisma.join. So, I built and issued a raw query manually.

Is there anything I missed here?

**Comment by austincrim** [comment, marked as answer]
Hey @MeoMix 👋,

Could you share your Prisma schema?

**Reply by MeoMix** [reply]
@austincrim Sure. Here's the relevant sections.

```
generator client {
  provider        = "prisma-client-js"
  binaryTargets   = ["native", "linux-musl"]
  previewFeatures = ["interactiveTransactions"]
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

model OrderItem {
  id                 String            @id @default(uuid()) @db.Uuid
  quantity           Int
  dateCreated        DateTime          @default(now()) @map("date_created")
  lastUpdated        DateTime          @updatedAt @map("last_updated")

  @@index([dateCreated(sort: Desc)])
  @@map("order_item")
}
```


OrderItem has a unique identifier which is a uuid. It maintains a last updated field. I'm performing a bulk update where I want to reject updating items which have changed since I last read their state.

**Reply by austincrim** [reply]
If you need to compare tuples, you will need to stick with `$queryRaw`. 

If you could split out the `id` and `lastUpdated` fields, you could issue a regular Prisma query that uses `AND` to check both conditions separately.

Feel free to open a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) for how Prisma could handle this scenario better.

**Comment by MeoMix** [comment]
Thank you for the confirmation, appreciate it.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
